Clarify `sink` behaviour in implicit subscription model (#680)
camaraproject/Commonalities: documentation/CAMARA-API-Event-Subscription-and-Notification-Guide.md

Reviewer: The attribute `sink` must be optional to not contradict the described behavior and the definition within artifacts (this inconsistency wasn't introduced by the PR, it pre-existed):

```suggestion
| sink           | string | https callback address where the notification must be POST-ed, `format: uri` SHOULD be used to require a string that is compliant with [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986).  The [security considerations](#43-notifications-security-considerations) SHOULD be followed.                | optional   |
```

Author: I will also update prior context (*), as `sink` was stated as mandatory in the context that is required to enable implicit subscriptions mode

Line 62:
If this capability is present in CAMARA API, the following attributes MUST be used in the POST request :

Author: Will be adding this note:

**NOTE**: To request the Implicit Subscription mode, `sink` attribute MUST be provided within the API request body.

Providing this capability is OPTIONAL for any CAMARA API depending on UC requirements.

> If this capability is present in CAMARA API, the following attributes MUST be used in the POST request:

Reviewer: With `sink` now optional, both rows of the table are optional, so "attributes MUST be used" reads as contradicting the cardinality column. Mandating the attribute *names* keeps the original intent:

```suggestion
If this capability is present in CAMARA API, the following attribute names MUST be used in the POST request:
```

| accessTokenExpireUtc | string date-time | An absolute UTC instant at which the access token shall be considered expired.    | optional    |
| accessTokenType      | string           | Type of access token - MUST be set to `bearer` for now                            | optional    |

**Sink delivery behaviour**

When `sink` is provided in the POST request, the following rules apply:

> - The API provider SHOULD support event delivery. If notification delivery is not supported (e.g. the implicit subscription feature is not implemented by this provider instance), the request MUST be rejected with `422 EVENT_NOTIFICATIONS_NOT_SUPPORTED` and the resource MUST NOT be created.

Reviewer 2: Clarification for error condition:

```suggestion
- API providers SHOULD implement event delivery support. If a request includes `sink` and the provider does not support notification delivery (i.e. the implicit subscription feature is not implemented by this provider instance), the request MUST be rejected with 422 EVENT_NOTIFICATIONS_NOT_SUPPORTED and the resource MUST NOT be created.
```

> - API providers SHOULD implement event delivery support. If a request includes `sink` and the API provider does not support notification delivery (i.e. the implicit subscription feature is not implemented by this provider instance), the request MUST be rejected with 422 EVENT_NOTIFICATIONS_NOT_SUPPORTED and the resource MUST NOT be created.

Reviewer: Vocabulary ... we haven't used "provider instance" yet in the document:

```suggestion
- API providers SHOULD implement event delivery support. If a request includes `sink` and the API provider does not support notification delivery (i.e. the implicit subscription feature is not implemented by this API provider), the request MUST be rejected with 422 EVENT_NOTIFICATIONS_NOT_SUPPORTED and the resource MUST NOT be created.
```

- When event delivery is accepted, `sink` MUST be echoed in the response.
- The implementation MUST NOT silently create the resource while ignoring the requested `sink`. If event delivery is unavailable, the consumer MUST be informed via the error response, so they can retry without `sink` (e.g. fall back to a polling pattern via `GET /<resource>`).

Aforemention rules imply a breaking change, so as it is worth to state below since Metarelease Sync26:
- New APIs that includes in their design the implicit subscription model MUST follow aforementioned behaviour.
- Previous existing APIs that owns in their design the implicit subscription model. Whether to document/align with this behaviour is an each API's own decision:
  - An API that wants to allow implementations without notification delivery adds it — that implies breaking change per section [7.4. Backward and Forward Compatibility of the CAMARA API Design Guide](./CAMARA-API-Design-Guide.md##74-backward-and-forward-compatibility) —, so a major version is required

>   - An API that expects every implementation to deliver events adds nothing and stays as it is.

Reviewer: Anchoring on the Commonalities release rather than the meta-release label, following the precedent in [Section 2.3. Strictness of Request Body Schemas](https://github.com/camaraproject/Commonalities/blob/main/documentation/CAMARA-API-Design-Guide.md#23-strictness-of-request-body-schemas) of the Design Guide ("The rule applies to CAMARA APIs targeting Commonalities r4.3 and later. APIs targeting earlier Commonalities versions are unaffected."). Also fixes the link to Section 7.4, which currently has a double `##` and so does not resolve, and makes explicit what an API that adds nothing expects from its implementations.

```suggestion
These rules apply to CAMARA APIs targeting Commonalities r4.4 and later. APIs targeting earlier Commonalities versions are unaffected.

- An API newly adopting the implicit subscription model MUST follow the behaviour described above.
- For an API that already offers the implicit subscription model, whether to document `422 EVENT_NOTIFICATIONS_NOT_SUPPORTED` is that API's own decision:
  - An API that allows implementations without notification delivery adds the error code to its API definition. This is a breaking change per [Section 7.4. Backward and Forward Compatibility](/documentation/CAMARA-API-Design-Guide.md#74-backward-and-forward-compatibility) of the CAMARA API Design Guide, so a new major version is required.
  - An API that expects every implementation to deliver event notifications adds nothing and stays as it is. Implementations of such an API MUST deliver event notifications when `sink` is provided.
```
